src/spikeinterface/core/generate.py

```diff
@@ -709,6 +716,12 @@ def synthesize_poisson_spike_vector(
     spike_frames[:num_correct_frames] = spike_frames[mask]  # Avoids a malloc
     unit_indices = unit_indices[mask]
 
+    # Add insertions
```

Comment:
As I mentioned before, I would like to keep this function simple. I don't want to add extra parameters and deal with the complexity that they bring.

I was suggestion to add these lines in your tests instead of adding an extra parameter to a low level function:

```python
        spike_frames, unit_indices = synthesize_poisson_spike_vector(**)
        spike_frames, unit_indices = _add_insertions(
            [spike_frames, unit_indices], insertions=insertions, insertions_replace=insertions_replace
        )
```

Or create another function called `synthesize_poisson_spike_vector_with_insertions` that calls this one and wraps your argument if you think that the function will be used a lot.

Am I missing something? Is there something important to gain by adding this? 
Fuctions with less parameters are easier to test, document and mantain and it seems that the win here will be saving one line of code writing.